oceanspy/tests/test_llc_rearrange.py

```diff
@@ -2837,3 +2886,1340 @@ def test_edge_arc_data(od, XRange, YRange, F_indx, Nx):
+@pytest.mark.parametrize(
+    "X, Y, Fs, exp",
+    [
+        ([45, 46], [89, 0], [1, 2], [46, 89]),
+        ([89, 0], [45, 44], [1, 4], [89, 44]),
+        ([45, 46], [0, 0], [1, 0], [46, 0]),
+        ([89, 0], [45, 40], [8, 9], [89, 40]),
+        ([0, 89], [45, 40], [8, 7], [0, 40]),
+        ([45, 40], [89, 0], [8, 11], [40, 89]),
+        ([89, 89], [89, 89], [0, 1], [None, None]),
+        ([0, 44], [45, 89], [1, 11], [0, 45]),
+        ([89, 40], [45, 9], [4, 8], [89, 49]),
+        ([45, 89], [0, 40], [8, 4], [49, 0]),
+        ([40, 0], [89, 40], [2, 6], [49, 89]),
+        ([0, 89], [40, 50], [7, 6], [0, 50]),
+        ([40, 70], [89, 0], [5, 6], [70, 89]),
+        ([0, 70], [40, 89], [10, 6], [0, 19]),
+        ([0, 40], [40, 89], [6, 2], [0, 49]),
+        ([70, 40], [0, 89], [6, 5], [40, 0]),
+        ([89, 0], [50, 40], [6, 7], [89, 40]),
+        ([70, 0], [89, 40], [6, 10], [49, 89]),
+    ],
+)
+def test_edge_slider(od, X, Y, Fs, exp):
+    face_connections = od.face_connections["face"]
+    if set([X[0], Y[0]]) == set([89]):
+        # upper right corner
+        with pytest.raises(ValueError):
+            edge_slider(X[0], Y[0], Fs[0], X[1], Y[1], Fs[1], face_connections)
+    else:
+        newP = edge_slider(X[0], Y[0], Fs[0], X[1], Y[1], Fs[1], face_connections)
+        assert newP == exp
+
+
+x1 = _np.array([k for k in range(15, 45)])
+y1 = [10] * len(x1)
+
+y2 = [k for k in range(45, 78)]
+x2 = len(y2) * [40]
+
+
+@pytest.mark.parametrize(
+    "x, y, facedir, ind, X0, X1",
+    [
+        (x1, y1, 1, -1, [x1[0], y1[0]], [89, y1[-1]]),
+        (x1, y1, 0, 0, [0, y1[0]], [x1[-1], y1[-1]]),
+        (x1, y1, 3, -1, [x1[0], y1[0]], [x1[-1], 89]),
+        (x1, y1, 2, 0, [x1[0], 0], [x1[-1], y1[-1]]),
+        (x2, y2, 2, 0, [x2[0], 0], [x2[-1], y2[-1]]),
+        (x2, y2, 0, -1, [x2[0], y2[0]], [0, y2[-1]]),
+        (x2, y2, 1, -1, [x2[0], y2[0]], [89, y2[-1]]),
+        (x2, y2, 0, 0, [0, y2[0]], [x2[-1], y2[-1]]),
+        (x2, y2, 1, -1, [x2[0], y2[0]], [89, y2[-1]]),
+    ],
+)
+def test_edge_completer(x, y, facedir, ind, X0, X1):
+    xn, yn = edge_completer(x, y, facedir, ind)
+    # assert unit distance between elements of array.
+    diffs = abs(_np.diff(xn)) + abs(_np.diff(yn))
+    assert _np.max(diffs) == _np.min(diffs) == 1
+    assert [xn[0], yn[0]] == X0
+    assert [xn[-1], yn[-1]] == X1
+
+
+# face = 0
+y0 = [k for k in range(45, 80)]
+x0 = len(y0) * [40]
+# face 1
+y11 = [k for k in range(5, 45)]
+x11 = len(y11) * [45]
+x12 = [k for k in range(45, 80)]
+y12 = len(x12) * [45]
+x1 = x11 + x12
+y1 = y11 + y12
+# face 4
+x41 = [k for k in range(5, 46)]
+y41 = len(x41) * [60]
+y42 = [k for k in range(59, 10, -1)]
+x42 = len(y42) * [45]
+x4 = x41 + x42
+y4 = y41 + y42
+# face 3
+y31 = [k for k in range(80, 40, -1)]
+x31 = len(y31) * [60]
+x32 = [k for k in range(60, 5, -1)]
+y32 = len(x32) * [40]
+x3 = x31 + x32
+y3 = y31 + y32
+# face 0 again!
+x01 = [k for k in range(80, 45, -1)]
```

Comment:
This look a little similar to that of test_subsample. I think it might be a good idea to use fixture, and have indirect = True in parametrize